### PR TITLE
Conditionally separate SITE_NAME and title with " - "

### PR DESCRIPTION
**Pull request by esoergel**
http://manage.dimagi.com/default.asp?176390
The problem is that title blocks aren't always set.
`<title>{% block title %}{% endblock title %}{{ SITE_NAME }}</title>`
looks good if it hasn't been overriden, but if it has, you get a title of `Edit Mobile WorkerCommCare HQ`
`<title>{% block title %}{% endblock title %} - {{ SITE_NAME }}</title>`
looks good if there IS a title, but looks like `- CommCare HQ` otherwise.

The easy answer is to set a default title, so we get `CommCare HQ - CommCare HQ`, but that's kinda ugly, and I found this solution pretty quickly with some googling.

I pulled in a djangosnippet template tag which allows you to grab bits of the template and assign them to a context variable.  This lets you see whether or not that block is overridden so you can conditionally insert the `-`
@biyeun

### CODE REVIEW

File: corehq/apps/style/templates/style/bootstrap2/base.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <html lang="{{ LANGUAGE_CODE }}">
     <head>
         <meta charset="utf-8"/>
-        <title>{% block title %}{% endblock title %}{{ SITE_NAME }}</title>
+        <title>{% block title %}{% endblock title %} - {{ SITE_NAME }}</title>
```
Comment (biyeun) [inline corehq/apps/style/templates/style/bootstrap2/base.html]: the homepage will show with just  - Commcare HQ, fyi

**Comment by esoergel** [comment]
Okay, this should get the job done.

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 2]
```diff
@@ -332,3 +332,32 @@ def render(self, context):
...
+
+class CaptureasNode(template.Node):
+    def __init__(self, nodelist, varname):
+        self.nodelist = nodelist
+        self.varname = varname
+
+    def render(self, context):
+        output = self.nodelist.render(context)
+        context[self.varname] = output
+        return ''
```
Comment (esoergel) [inline corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: This whole thing is copypasta'd from https://djangosnippets.org/snippets/545/ except for the docstring.  I googled this problem a bit and found people recommending that.

**Comment by proteusvacuum** [comment]
neat! looks like something similar to `captureas` is going to be in django 1.9
https://code.djangoproject.com/ticket/21695

**Comment by esoergel** [comment]
That's cool, I wonder what limitations there will be on what can be captured.  There was a ticket for this specific thing here: https://code.djangoproject.com/ticket/6378

**Pull request merged**